Streamlining JavaScript house style (#268)
springernature/frontend-playbook: javascript/house-style.md

We use white space liberally to help keep code readable. You're encouraged to use newlines to break up long functions into logical chunks.

You should also remove trailing white space from lines of code. Your editor should be able to either remove this or highlight it when present, for example, [atom-whitespace](https://github.com/atom/whitespace), [vscode-trailing-spaces](https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces).

### Variables

#### Use of const and let
We define variables with multiple `const` and `let` statements.

We do this:

```js
const foo = 1;
const bar = 2;
```

We _don't_ do this:

```js
const foo = 1,
      bar = 2;
```

> We use `let` only when a variable _explicitly_ [needs to be mutable](https://ada.is/blog/2015/07/13/immutable/):

Reviewer: Is there a missing example here? The colon on the end suggests so 😉

Author: Ah yes, I found the example given a bit meaningless.

```js	
let foo = 'bar';	
```

Author: Maybe this is an ok example?

We do this:

```js
let foo = 1;

if (foo) {	
    foo += 1;	
}
```

```js
const foo = [];

if (bar) {
    foo.push(1);
}
```

We _don't_ do this:

```js
let foo = [];

if (bar) {
    foo.push(1);
}
```

Reviewer: maybe using a loop would be a good way to use a more meaningful example? e.g.

```
for(let i = 0; i < 10; i++)  { }
```

I'm not wedded to this suggestion, though - I'm only making it because it would make sense to have an example if `const` and `let` had their own headings! But they don't _need_ separate headings, especially if only one has a meaningful code example 🤷‍♀️

Reviewer: oh you beat me to the comment 😄 

Yeah, i like those examples!

Author: I actually changed the heading to "Defining variables".
What do you think? https://github.com/springernature/frontend-playbook/pull/268/commits/356330ab79cfc15e0d0e3ba5d282c4111643768f

Reviewer: yeah, nicely done 👍

#### Use of var

> If the environment you're running in does not support ES2015 (Node.js 4.x/[Babel](https://babeljs.io/)), and therefore you are using `var` instead of `const` and `let`, we trust our developers to understand [hoisting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#var_hoisting). 

Reviewer: This sentence is a bit tortured. I know it's not your original work though, so you are blameless 😄 

```suggestion
If you have to use `var` instead of `const` and `let` (because your environment doesn't support ES2015 (Node.js 4.x/[Babel](https://babeljs.io/))), you need to understand [hoisting](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var#var_hoisting). 
```

> We generally discourage defining variables inside loops or blocks.

Reviewer: ```suggestion
Avoid defining variables inside loops or blocks.
```

(makes the instruction stronger ✊)
